presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/MetastoreUriFetcher.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore.thrift;
+
+import com.facebook.airlift.http.client.HttpClient;
+import com.facebook.airlift.http.client.Request;
+import org.apache.thrift.TException;
+
+import java.net.URI;
+
+public interface MetastoreUriFetcher
```

Comment:
Why not make `MetastoreUriFetcher` to inherit `ThriftMetastoreUriFetcher`. Then we can remove this interface.

Reply:
This interface was added because a MockMetastoreUriFetcher is required for unit tests. The actual implementation and the mock both implement the ThriftMetastoreUriFetcher. Mock - https://github.com/prestodb/presto/pull/15694/files#diff-f9fbf6a66f086e2a385ed6efbe6b856c19bdf594934ac7d1007d3b7d9346479c

Reply:
I also added 2 more mock MetastoreUriFetcher's to unit test scenarios like empty uri returned, malformed uri returned, uri returned with non-thrift schema, etc. Do let me know if there is a better way to do this.

Reply:
There is no need to use interface the unit tests. Let's remove this interface and directly use `ThriftMetastoreUriFetcher`. Then for your mock classes, just directly inherit `ThriftMetastoreUriFetcher`. Because you only create new instances of your mocks through `::new`